core/src/tile/labels/labelContainer.cpp

```diff
@@ -34,6 +41,8 @@ bool LabelContainer::addLabel(MapTile& _tile, const std::string& _styleName, Lab
 }
 
 void LabelContainer::updateOcclusions() {
+    m_currentZoom = m_view->getZoom();
```

Comment:
We can pass m_view as a reference here instead of storing a shared_ptr for m_view in labelContainer.